### PR TITLE
Added the fileuploader component.

### PR DESCRIPTION
**Pull request by AyushNautiyalDeveloper**

### CODE REVIEW

File: components/file-upload/README.md
Comment (ShreyDhyani) [inline components/file-upload/README.md:1]: Remove

File: components/file-upload/src/HeadLessFileUpload.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+}
+
+function HeadLessFileUpload(
+	props: HeadLessFileUploadProps,
+	ref: ForwardedRef<HTMLInputElement>,
+) {
+	const {
+		handleAfterFileUploadHook,
+		acceptedFileTypes,
+		allowsMultiple,
```
Comment (ShreyDhyani) [inline components/file-upload/src/HeadLessFileUpload.tsx:18]: Can pass default values here: 
```suggestion
		allowsMultiple = false,
```

File: components/file-upload/src/ReuseFileUpload.tsx [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+					></div>
+				)
+			) : (
+				<>
+					{showChildren ? childrenWithProps : <></>}
+					<HeadLessFileUpload
+						ref={ref}
+						className={twMerge("hidden " + baseClassName)}
+						handleAfterFileUploadHook={prepareOnChange}
+						{...rest}
```
Comment (ShreyDhyani) [inline components/file-upload/src/ReuseFileUpload.tsx:167]: What are these props??

File: components/input-group/rollup.config-1701871668842.cjs
Comment (ShreyDhyani) [inline components/input-group/rollup.config-1701871668842.cjs:1]: Remove

File: components/modal/rollup.config-1701858730830.cjs
Comment (ShreyDhyani) [inline components/modal/rollup.config-1701858730830.cjs:1]: remove

File: development/file-upload-app/pages/file-upload/index.mdx [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+import MultiFileReuseFileUpload from "../../components/file-upload/MultiFileReuseFileUpload/MultiFileReuseFileUpload"
+import AcceptFileUpload from "../../components/file-upload/AcceptFileUpload/AcceptFileUpload"
+
+## File Upload
+
+## Overview
+
+The `HeadlessFileUpload` component is a versatile File Upload component for React applications. It allows for File Uploading and Provides Custom Implemnetation Possiblilties.
+
+
```
Comment (ShreyDhyani) [inline development/file-upload-app/pages/file-upload/index.mdx:20]: Add 
## Installation

File: development/file-upload-app/pages/file-upload/index.mdx [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+Import the `ReuseFileUpload` component into your React application as follows:
+
+```jsx
+import { ReuseFileUpload  } from "@locoworks/reusejs-react-file-upload";
+```
+
+
+## Types/ Exported Components
+
+The component exports the following types and components:
```
Comment (ShreyDhyani) [inline development/file-upload-app/pages/file-upload/index.mdx:52]: Remove types

File: docs/pages/fileupload/index.mdx
Comment (ShreyDhyani) [inline docs/pages/fileupload/index.mdx:1]: Remove

File: yarn.lock [hunk 5]
```diff
@@ -5497,6 +5497,21 @@ listr2@^5.0.7:
     through "^2.3.8"
     wrap-ansi "^7.0.0"
 
+livereload-js@^3.3.1:
```
Comment (ShreyDhyani) [inline yarn.lock:5500]: Remove

File: yarn.lock [hunk 6]
```diff
@@ -7617,6 +7637,13 @@ rimraf@^3.0.2:
   dependencies:
     glob "^7.1.3"
 
+rollup-plugin-livereload@^2.0.5:
```
Comment (ShreyDhyani) [inline yarn.lock:7640]: Remove

File: yarn.lock [hunk 7]
```diff
@@ -8896,6 +8923,11 @@ wrappy@1:
   resolved "https://registry.yarnpkg.com/wrappy/-/wrappy-1.0.2.tgz#b5243d8f3ec1aa35f1364605bc0d1036e30ab69f"
   integrity sha512-l4Sp/DRseor9wL6EvV2+TuQn63dMkPjZ/sp9XkghTEbV9KlPS1xUsZ3u7/IQO4wxtcFB4bgpQPRcR3QCvezPcQ==
 
+ws@^7.4.3:
```
Comment (ShreyDhyani) [inline yarn.lock:8926]: Why is this package added?

File: yarn.lock [hunk 8]
```diff
@@ -6601,6 +6616,11 @@ optionator@^0.9.1, optionator@^0.9.3:
     prelude-ls "^1.2.1"
     type-check "^0.4.0"
 
+"opts@>= 1.2.0":
```
Comment (ShreyDhyani) [inline yarn.lock:6619]: What is this package?

File: components/file-upload/src/HeadLessFileUpload.tsx [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+
+export interface HeadLessFileUploadProps {
+	acceptedFileTypes?: Array<string>;
+	allowsMultiple?: boolean;
+	handleAfterFileUploadHook?: (files: FileList | null) => Promise<void>;
+	children?: React.ReactNode;
+	className?: string;
+}
+
+function HeadLessFileUpload(
```
[the remark this answers is not in the document]
Comment (AyushNautiyalDeveloper) [reply components/file-upload/src/HeadLessFileUpload.tsx:11]: done.

File: components/file-upload/README.md
Comment (AyushNautiyalDeveloper) [reply components/file-upload/README.md:1]: removed.

File: components/file-upload/src/HeadLessFileUpload.tsx [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+		: undefined;
+
+	return (
+		<input
+			type="file"
+			ref={ref}
+			className={className}
+			accept={acceptAttribute}
+			onChange={async (e: any) => {
+				handleAfterFileUploadHook?.(e);
```
[the remark this answers is not in the document]
Comment (AyushNautiyalDeveloper) [reply components/file-upload/src/HeadLessFileUpload.tsx:34]: added

File: components/file-upload/src/HeadLessFileUpload.tsx (hunk 9 above)
[the remark this answers is not in the document]
Comment (AyushNautiyalDeveloper) [reply components/file-upload/src/HeadLessFileUpload.tsx:11]: added

File: components/file-upload/README.md
Comment (AyushNautiyalDeveloper) [reply components/file-upload/README.md:1]: removed

File: components/file-upload/src/HeadLessFileUpload.tsx (hunk 1 above)
Comment (AyushNautiyalDeveloper) [reply components/file-upload/src/HeadLessFileUpload.tsx:18]: passed

File: components/file-upload/src/ReuseFileUpload.tsx (hunk 2 above)
Comment (AyushNautiyalDeveloper) [reply components/file-upload/src/ReuseFileUpload.tsx:167]: they are the native properties of the html input element that can be passed.

File: components/input-group/rollup.config-1701871668842.cjs
Comment (AyushNautiyalDeveloper) [reply components/input-group/rollup.config-1701871668842.cjs:1]: removed.

File: components/modal/rollup.config-1701858730830.cjs
Comment (AyushNautiyalDeveloper) [reply components/modal/rollup.config-1701858730830.cjs:1]: removed.

File: development/file-upload-app/pages/file-upload/index.mdx (hunk 4 above)
Comment (AyushNautiyalDeveloper) [reply development/file-upload-app/pages/file-upload/index.mdx:52]: ok

File: yarn.lock (hunk 5 above)
Comment (AyushNautiyalDeveloper) [reply yarn.lock:5500]: removed.

File: yarn.lock (hunk 6 above)
Comment (AyushNautiyalDeveloper) [reply yarn.lock:7640]: removed.

File: components/file-upload/src/ReuseFileUpload.tsx (hunk 2 above)
Comment (AyushNautiyalDeveloper) [reply components/file-upload/src/ReuseFileUpload.tsx:167]: they are the props of the input elment user may want to pass the native props of the input. like id="uniqueId"

File: components/input-group/rollup.config-1701871668842.cjs
Comment (AyushNautiyalDeveloper) [reply components/input-group/rollup.config-1701871668842.cjs:1]: removed

File: yarn.lock (hunk 5 above)
Comment (AyushNautiyalDeveloper) [reply yarn.lock:5500]: removed.

File: yarn.lock (hunk 6 above)
Comment (AyushNautiyalDeveloper) [reply yarn.lock:7640]: removed.

File: yarn.lock (hunk 8 above)
Comment (AyushNautiyalDeveloper) [reply yarn.lock:6619]: didnt added it externally may be it was added with the live reload.

File: yarn.lock (hunk 7 above)
Comment (AyushNautiyalDeveloper) [reply yarn.lock:8926]: added with the live reload.

**Review by ShreyDhyani — APPROVED** [review]

**Pull request merged**